### ISSUE TITLE
ssl check "on_broken" option not working

### ISSUE THREAD

**Issue by fjomier** [StatusCakeDev/terraform-provider-statuscake#82]
**Describe the bug**
Hello StatusCakeDev team,

First, I'd like to thank you for this terraform provider, it's really helpful to use your product.

Here's an annoying bug I faced when trying to switch to your terraform provider and your new v1 API.

When creating a ssl check with the option `on_broken` set to `true`, it's not taken into account.
When trying to apply a second time , the terraform operations fails.

**To Reproduce**

XXXXXX is the exact same redacted value everywhere.

1. Create the ssl check

```
% terraform apply
[...]

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # statuscake_ssl_check.bug will be created
  + resource "statuscake_ssl_check" "bug" {
      + check_interval   = 86400
      + follow_redirects = false
      + id               = (known after apply)
      + paused           = false

      + alert_config {
          + alert_at    = [
              + 1,
              + 7,
              + 30,
            ]
          + on_broken   = true
          + on_expiry   = true
          + on_mixed    = false
          + on_reminder = true
        }

      + monitored_resource {
          + address = "https://www.google.com"
        }
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

statuscake_ssl_check.bug: Creating...
statuscake_ssl_check.bug: Creation complete after 1s [id=XXXXXX]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.
```

2. Check the ssl check state : 
```
% curl -s -H "Authorization: Bearer ${SC_API_KEY}" 'https://api.statuscake.com/v1/ssl/XXXXXX' | jq '.'
{
  "data": {
    "id": "XXXXXX",
    "paused": false,
    "check_rate": 86400,
    "website_url": "https://www.google.com",
    "issuer_common_name": "",
    "cipher": "",
    "cipher_score": 0,
    "certificate_score": 0,
    "certificate_status": "",
    "contact_groups": [],
    "alert_at": [
      1,
      7,
      30
    ],
    "last_reminder": 0,
    "alert_reminder": true,
    "alert_expiry": true,
    "alert_broken": false, <== the problem is here
    "alert_mixed": false,
    "follow_redirects": false,
    "mixed_content": [],
    "flags": {
      "is_extended": false,
      "has_pfs": true,
      "is_broken": false,
      "is_expired": false,
      "is_missing": false,
      "is_revoked": false,
      "has_mixed": false,
      "follow_redirects": false
    }
  }
}
```

3. Try another apply

```
% terraform apply
[...]
statuscake_ssl_check.bug: Refreshing state... [id=XXXXXX]

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # statuscake_ssl_check.bug will be updated in-place
  ~ resource "statuscake_ssl_check" "bug" {
        id               = "XXXXXX"
        # (4 unchanged attributes hidden)

      ~ alert_config {
          ~ on_broken   = false -> true
            # (4 unchanged attributes hidden)
        }

        # (1 unchanged block hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

statuscake_ssl_check.bug: Modifying... [id=XXXXXX]
╷
│ Error: failed to update SSL check: Unexpected error. Try again later.
│
│   with statuscake_ssl_check.bug,
│   on main.tf line 39, in resource "statuscake_ssl_check" "bug":
│   39: resource "statuscake_ssl_check" "bug" {
│
╵
```

**Expected behavior**
`alert_broken` should be set to `true`

**Additional informations**
```
terraform {
  required_providers {
    statuscake = {
      source  = "StatusCakeDev/statuscake"
      version = "2.0.3"
    }
  }
  required_version = "~>1.2.5"
}
```

```
% terraform version
Terraform v1.2.5
on darwin_amd64
+ provider registry.terraform.io/statuscakedev/statuscake v2.0.3
```

**Comment by tomasbasham** [comment]
Thanks for reporting this. It appears there is a typo in the SSL resource. A patch has been made to fix this

**Comment by macropin** [comment]
Thanks for releasing this fix! 2.0.4 is working for us now.
